Lib/_pydecimal.py

```diff
@@ -500,8 +504,15 @@ def sin(x):
     >>> print(getcontext().prec)
     28
     """
-    if ctx is None: ctx = getcontext()
-    return _ContextManager(ctx)
+    if ctx is None:
+        ctx = getcontext()
+    ctx_manager = _ContextManager(ctx)
+    with ctx_manager as ctx:
```

Comment:
Using `with` here is a bit weird; it sets and unsets the active context unnecessarily. Can you just use `ctx_manager.new_context` directly?

(Also, thanks @TeamSpen210 for catching this!)